graph/api/src/main/java/org/jboss/windup/graph/service/ProjectService.java

```diff
@@ -67,4 +78,62 @@ public ProjectModel getOrCreateSharedLibsProject()
 
         return sharedLibsProject;
     }
+
+    public Map<ProjectModel, ProjectModel> getProjectToRootProjectMap()
+    {
+        Map<ProjectModel, ProjectModel> projectModels = new HashMap<>();
+
+        for (FileModel inputPath : WindupConfigurationService.getConfigurationModel(this.getGraphContext()).getInputPaths())
+        {
+            ProjectModel rootProjectModel = inputPath.getProjectModel();
+            if (rootProjectModel == null)
+            {
+                continue;
+            }
+
+            ProjectModelTraversal traversal = new ProjectModelTraversal(rootProjectModel, new OnlyOnceTraversalStrategy());
```

Comment:
I'm not sure, this is based on this code: https://github.com/windup/windup-web/blob/master/addons/web-support/impl/src/main/java/org/jboss/windup/web/addons/websupport/rest/MigrationIssuesEndpointImpl.java#L72

from MigrationIssues.